### DISCUSSION TITLE
Path Parameter

### DISCUSSION THREAD

**Discussion by sindhubha** [fastapi/fastapi#6081] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@app.post("/get_company_list/{company_id}")
async def get_company_list(
        company_id: str = Path(default=None, required=False),
```

### Description

is there any option in path parameter and require=false, i want company_id is path parameter and also company_id is not manatory. if it is posible pls give me the solution

### Operating System

Windows

### FastAPI Version

0.92

### Python Version

3.10.0

**Comment by sindhubha** [comment]
is there any option In OpenAPI, path parameters are not required from fastapi .

**Comment by ebottos94** [comment, marked as answer]
Hi @sindhubha, 

you can do something like this : 

```python
from fastapi import FastAPI, Query
from typing import Optional

app = FastAPI()


@app.post("/get_company_list/{company_id}")
@app.post("/get_company_list")
async def get_company_list(company_id: Optional[str] = Query(None)) :
    if company_id :
        return company_id
    else :
        return "not company_id"
```

**Reply by ebottos94** [reply]
@sindhubha if your problem is solved can you mark the comment as the answer ? Thank you.

**Reply by vincentsarago** [reply]
FYI this is not possible after Fastapi 0.95 see https://github.com/tiangolo/fastapi/discussions/9278

**Reply by ebottos94** [reply]
I've reply in your discussion, simply declare path parameter as str of None type and assign default or None value to it.
